configuration/cli/envchk/envchk.cpp

```diff
@@ -0,0 +1,341 @@
+                    }
+                    else if (lvl == "w")
+                    {
+                        outputLevel = statusMsg::warning;
+                    }
+                    else if (lvl == "i")
+                    {
+                        outputLevel = statusMsg::info;
+                    }
+                    else
+                    {
+                        throw CliException("Output level is not valid");
+                    }
+                }
+            }
+
+            else if (optName == "-v" || optName == "--verbose")
+            {
+                verbose = true;
+            }
+
+            else if (idx == argc)
+            {
+                envFile = optName;
+            }
+
+            else
+            {
+                throw CliException("Parameters are incorrect");
+            }
+        }
+    }
+    catch(const CliException &e)
+    {
+        std::cout << "There was an issue processing options: " << e.what() << std::endl;
+        rc = false;
+    }
+    return rc;
+}
+
+
+bool validate()
+{
+
+    if (!dirExists(configSchemaDir))
+    {
+        std::cout << "Schema directory " << configSchemaDir << " does not exist" << std::endl;
+        return false;
+    }
+
+    if (!dirExists(configSchemaPluginsDir))
+    {
+        std::cout << "Schema plugins directory " << configSchemaPluginsDir << " does not exist" << std::endl;
+        return false;
+    }
+
+    if (!fileExists(configSchemaDir + masterSchemaFile))
+    {
+        std::cout << "The master config file " << masterSchemaFile << " does not exist" << std::endl;
+        return false;
+    }
+
+    if (!envFile.empty())
+    {
+        if (!fileExists(envFile))
+        {
+            std::cout << "The environment file " << envFile << " does not exist" << std::endl;
+            return false;
+        }
+    }
+
+    return true;
+}
+
+
+std::string getNextArg(int argc, char *argv[], int idx)
+{
+    if (idx < argc)
+    {
+        return std::string(argv[idx]);
+    }
+    throw CliException("Arguments exhausted when more expected");
+}
+
+
+bool dirExists(const std::string &dirName)
+{
+    bool rc = true;
+    struct stat info;
+    if (stat(dirName.c_str(), &info) != 0)
+    {
+        rc = false;
+    }
+    rc = ((info.st_mode&S_IFMT)==S_IFDIR);
+    return rc;
+}
+
+
+bool fileExists(const std::string &fileName)
+{
+    struct stat info;
+    return stat(fileName.c_str(), &info) == 0;
+}
+
+
+std::vector<std::string> splitString(const std::string &input, const std::string &delim)
+{
+    size_t  start = 0, end = 0, delimLen = delim.length();
+    std::vector<std::string> list;
+
+    while (end != std::string::npos)
+    {
+        end = input.find(delim, start);
+        std::string item = input.substr(start, (end == std::string::npos) ? std::string::npos : end - start);
+        if (!item.empty())
+            list.push_back(item);
+        start = ((end > (std::string::npos - delimLen)) ? std::string::npos : end + delimLen);
```

Comment:
This seems incorrect. ``end > (std::string::npos - delimLen)`` I think this would always be true. End is always >= -1 and (-1 - delimLen) should always be less than -1

Reply:
It would seem this is the case, however the code will successfully parse and split the string. The size_t is unsigned, so the test really looks at end being the case where the input.find did not find a match. In this case it just assigns start so that the next find will return npos. However, since end is already npos, it will exit the loop anyway. This could be cleaner. Probably as simple check using the length of the input.